imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/CCITTFaxDecoderStream.java

```diff
@@ -356,6 +356,9 @@ private int decodeRun(final Tree tree) throws IOException {
             if (n.isLeaf) {
                 total += n.value;
                 if (n.value < 64) {
+                    if (n.value < 0) {
```

Comment:
Fair enough. 
I guess then we could re-arrange the branches so that the normal cases (n.value >= 64) is handled before the terminating codes, and still avoid the extra nesting?